Purchases/Purchasing/StoreKitAbstractions/StoreProductDiscount.swift

```diff
@@ -135,8 +139,9 @@ extension StoreProductDiscount {
     }
 
     @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
-    internal convenience init?(sk2Discount: SK2ProductDiscount) {
-        guard let discount = SK2StoreProductDiscount(sk2Discount: sk2Discount) else { return nil }
+    internal convenience init?(sk2Discount: SK2ProductDiscount, currencyCode: String?) {
```

Comment:
This is the only part of this PR that feels weird to me, why does the discount need the currency code too?

Reply:
This also felt weird 😬 Hybrid common looks for currency code on a discount. And because it has a price on the object but no current code (right now) so it does _kind_ of felt like it should have one?

The JSON representation of a SK2 product has the currency code for the offers in it so that makes it kind of seem like this should also have a currency code?

If this isn't great we can totally rework hybrid common but I'm not expert here yet 🤷‍♂️ 

cc: @aboedo

Reply:
> The JSON representation of a SK2 product has the currency code for the offers in it so that makes it kind of seem like this should also have a currency code?

Ok that makes sense then. This is just an implementation detail so it's not a big deal, and it does provide more data to users so I'm good with it.

Reply:
Ideally there should be a feedback/radar filed for adding `jsonRepresentation` to `StoreKit.Product.SubscriptionOffer` like how `StoreKit.Product` has? I think... 🤷‍♂️